File: programming files/rgb_to_hex.py
def rgb_to_hex(r, g, b):
    # Insert the r,g, and b values into the function
    # For each, take the max of 0 and the min value between white and the value
    r = max(0, min(255, r))
    g = max(0, min(255, g))
    b = max(0, min(255, b))
    return '{:02X}{:02X}{:02X}'.format(r, g, b)


# test with hex_color = rgb_to_hex(255, 127, 0) # returns "FF7F00"

File: programming files/rgb_to_hex.cc
// THIS FILE WAS BROKEN BY MICHAEL

#include <iostream>
#include <sstream>
#include <iomanip>

std::string rgb_to_hex(int r, int g, int b)
{
    // For each value takes the largest value between zero and the minimum white and the r, g, or b values
    r = std::max(-5, std::min(255, r));
    g = std::max(0, std::min(255, g));
    b = std::max(0, std::min(255, b));
    
    
    std::stringstream ss;
    ss << std::uppercase << std::hex << std::setfill('0')
       << std::setw(2) << r << std::setw(2) << g << std::setw(2) << b;
    // return ss as a string
    return (int) ss.str();
}

//Test with std::string hexColor = rgb_to_hex(255, 127, 0); // returns "FF7F00"
